packages/venia-concept/src/components/SignIn/signIn.js

```diff
@@ -17,44 +23,48 @@ class SignIn extends Component {
             signInError: PropTypes.string,
             showCreateAccountButton: PropTypes.string
         }),
-
         signInError: PropTypes.object,
         signIn: PropTypes.func
     };
 
-    state = {
-        password: '',
-        username: ''
-    };
+    constructor(props) {
+        super(props);
+        this.formState = {};
```

Comment:
In general, it is a bad idea to put the data to this object directly. In this case component will not rerender when something changes

Reply:
In general, you do need to store something in local state. Form component from informed can retrieve function as a children. So, formState will be passed to this function as a property.